contracts/core/EntryPoint.sol

```diff
@@ -121,9 +122,14 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuard,
             }
             // handleOps was called with gas limit too low. abort entire bundle.
             if (innerRevertCode == INNER_OUT_OF_GAS) {
-                //report paymaster, since if it is not deliberately caused by the bundler,
-                // it must be a revert caused by paymaster.
+                //can only be caused by bundler (leaving not enough gas for inner call)
                 revert FailedOp(opIndex, "AA95 out of gas");
+            } else if (innerRevertCode == INNER_REVERT_LOW_PREFUND) {
+                uint256 actualGas1 = preGas - gasleft() + opInfo.preOpGas;
+                uint256 prefund = opInfo.prefund;
```

Comment:
I think this optimisation is not needed, this branch is pretty exotic, don't you think?